src/include/platform/internal/GenericPlatformManagerImpl.cpp

```diff
@@ -288,6 +299,27 @@ void GenericPlatformManagerImpl<ImplClass>::HandleMessageLayerActivityChanged(bo
     }
 }
 
+template <class ImplClass>
+void GenericPlatformManagerImpl<ImplClass>::HandleDeviceRebooted(intptr_t arg)
+{
+    PlatformManagerDelegate * platformManagerDelegate       = PlatformMgr().GetDelegate();
+    GeneralDiagnosticsDelegate * generalDiagnosticsDelegate = GetDiagnosticDataProvider().GetGeneralDiagnosticsDelegate();
+
+    if (generalDiagnosticsDelegate != nullptr)
+    {
+        generalDiagnosticsDelegate->OnDeviceRebooted();
```

Comment:
Could you also remove this call from `platform/Zephyr/PlatformManagerImpl.cpp`?

Reply:
Acked